src/storage/table.cc

```diff
@@ -66,6 +69,26 @@ Table::Table(const std::string& name, uint32_t id, uint32_t pid, uint64_t ttl, b
     AddVersionSchema(*table_meta_);
 }
 
+Table* Table::CreateTable(const std::string& name, uint32_t id, uint32_t pid, uint32_t seg_cnt,
+              const std::map<std::string, uint32_t>& mapping, uint64_t ttl,
+              ::openmldb::type::TTLType ttl_type,
+              const std::string& db_root_path,
+              ::openmldb::common::StorageMode storage_mode) {
+    if (storage_mode == ::openmldb::common::StorageMode::kMemory) {
+        return new MemTable(name, id, pid, seg_cnt, mapping, ttl, ttl_type);
+    } else {
+        return new DiskTable(name, id, pid, mapping, ttl, ttl_type, storage_mode, db_root_path);
+    }
+}
+
+Table* Table::CreateTable(const ::openmldb::api::TableMeta& table_meta, const std::string& db_root_path) {
+    if (table_meta.storage_mode() == ::openmldb::common::StorageMode::kMemory) {
+        return new MemTable(table_meta);
+    } else {
+        return new DiskTable(table_meta, db_root_path);
```

Comment:
- `db_root_path` is a gflag variable, need not pass from here
- Where the function will be called

Reply:
* for question1: The two `db_root_path` are not the same. The gflag one is like `/tmp/`. It's used for the location for snapshot of memtable. It's passed from tests and nameserver to tablet. Here this `db_root_path` is like `/tmp/1_1`. It's the detailed location for one tid and one pid. For a disktable or a snapshot.  It's passed from tablet to table. 
* for question2: In this PR, it's called in table_test to provide a unified way to create a table and run tests. And in the next PR(unfinished), it will be called by Tablet::CreateTable.

Reply:
1 you can use ssd_root_path/hdd_root_path directly in disktable 
2 move  function to test file.

Reply:
I think passing `db_root_path` here is actually confusing. And the factory method is not very straightforward, since mem table does not require the `path` parameter. Why not:
- remove these two factory methods. Just use the `new DiskTable/MemTable` directly
- For `DiskTable`, no need to pass the `db_root_path`. Just pass the `table_path`, which is derived from either `hdd_root_path` or `ssd_root_path`.